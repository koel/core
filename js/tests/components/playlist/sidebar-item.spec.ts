import Component from '@/components/playlist/sidebar-item.vue'
import NameEditor from '@/components/playlist/name-editor.vue'
import factory from '@/tests/factory'
import { shallow, mount } from '@/tests/adapter'

describe('components/playlist/sidebar-item', () => {
  let playlist: Playlist
  beforeEach(() => {
    playlist = factory<Playlist>('playlist', {
      id: 99,
      name: 'Foo'
    })
  })

  afterEach(() => {
    jest.resetModules()
  })

  it('renders a playlist menu item', () => {
    const wrapper = shallow(Component, {
      propsData: { playlist }
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the Favorites menu item', () => {
    const wrapper = shallow(Component, {
      propsData: {
        playlist: {
          name: 'Favorites'
        },
        type: 'favorites'
      }
    })
    expect(wrapper.find('li.playlist.favorites').text()).toMatch('Favorites')
  })

  // skipping because buggy test utils
  it.skip('edits a playlist', async () => {
    const wrapper = mount(Component, {
      propsData: { playlist }
    })

    wrapper.dblclick('li.playlist:last-of-type:not(.favorites):not(.recently-played)')
    await wrapper.vm.$nextTick()
    expect(wrapper.has(NameEditor)).toBe(true)
  })

  // skipping because buggy test utils
  it.skip("doesn't allow editing Favorites item", async () => {
    const wrapper = shallow(Component, {
      propsData: {
        playlist: { name: 'Favorites' },
        type: 'favorites'
      }
    })

    wrapper.dblclick('li.favorites')
    await wrapper.vm.$nextTick()
    expect(wrapper.has(NameEditor)).toBe(false)
  })
})
