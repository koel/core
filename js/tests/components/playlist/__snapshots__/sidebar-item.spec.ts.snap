// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/playlist/sidebar-item renders a playlist menu item 1`] = `
<li class="playlist playlist"><a href="#!/playlist/99" class="">Foo</a>
  <!---->
  <contextmenu-stub playlist="[object Object]"></contextmenu-stub>
</li>
`;
